File: MstGrid/VB Demo Project/VBDemo/frmColumn.vb

Imports System.Drawing

Friend Class frmColumn


    Private Sub frmColumn_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        'Get property value
        If frmDemo.GrdView1.FixedRows > 0 Then
            txtColumnTitle.Text = frmDemo.GrdView1.Cell(0, frmDemo.GrdView1.CurrentColIndex).Text
            txtColumnTitle.Enabled = True
            txtColumnTitle.BackColor = ColorTranslator.FromOle(&HFFFFFF)
        Else
            txtColumnTitle.Text = ""
            txtColumnTitle.Enabled = False
            txtColumnTitle.BackColor = ColorTranslator.FromOle(&H8000000F)
        End If
        txtColumnFormat.Text = frmDemo.GrdView1.Column(frmDemo.GrdView1.CurrentColIndex).FormatString
        txtMaxLength.Text = CStr(frmDemo.GrdView1.Column(frmDemo.GrdView1.CurrentColIndex).MaxLength)
        txtDecimalLength.Text = CStr(frmDemo.GrdView1.Column(frmDemo.GrdView1.CurrentColIndex).DecimalLength)

    End Sub


    Private Sub cmdOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdOK.Click

        frmDemo.GrdView1.AutoRedraw = False
        If txtColumnTitle.Enabled Then
            frmDemo.GrdView1.Cell(0, frmDemo.GrdView1.CurrentColIndex).Text = txtColumnTitle.Text
        End If
        frmDemo.GrdView1.Column(frmDemo.GrdView1.CurrentColIndex).FormatString = txtColumnFormat.Text
        frmDemo.GrdView1.Column(frmDemo.GrdView1.CurrentColIndex).MaxLength = Val(txtMaxLength.Text)
        frmDemo.GrdView1.Column(frmDemo.GrdView1.CurrentColIndex).DecimalLength = Val(txtDecimalLength.Text)
        frmDemo.GrdView1.AutoRedraw = True
        Me.Close()

    End Sub


    Private Sub cmdCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdCancel.Click

        Me.Close()

    End Sub


End Class

File: MstGrid/C# Demo Project/CSDemo/frmColumn.cs
using System.Drawing;
using System.Windows.Forms;
using System;



partial class frmColumn
{
    public frmColumn()
    {
        InitializeComponent();

        //Added to support default instance behavour in C#
        if (defaultInstance == null)
            defaultInstance = this;
    }

    #region Default Instance

    private static frmColumn defaultInstance;

    public static frmColumn Default
    {
        get
        {
            if (defaultInstance == null)
            {
                defaultInstance = new frmColumn();
                defaultInstance.FormClosed += new FormClosedEventHandler(defaultInstance_FormClosed);
            }

            return defaultInstance;
        }
    }

    static void defaultInstance_FormClosed(object sender, FormClosedEventArgs e)
    {
        defaultInstance = null;
    }

    #endregion


    private void frmColumn_Load(object sender, EventArgs e)
    {
        //Get property values
        if (frmDemo.Default.GrdView1.FixedRows > 0)
        {
            txtColumnTitle.Text =Convert.ToString ( frmDemo.Default.GrdView1.Cell(0, frmDemo.Default.GrdView1.CurrentColIndex).Text) ;
            txtColumnTitle.Enabled = true;
            txtColumnTitle.BackColor = ColorTranslator.FromOle(0xFFFFFF);
        }
        else
        {
            txtColumnTitle.Text = "";
            txtColumnTitle.Enabled = false;
            txtColumnTitle.BackColor = ColorTranslator.FromOle(0x800000);
        }
        txtColumnFormat.Text = frmDemo.Default.GrdView1.Column(frmDemo.Default.GrdView1.CurrentColIndex).FormatString;
        txtMaxLength.Text = frmDemo.Default.GrdView1.Column(frmDemo.Default.GrdView1.CurrentColIndex).MaxLength.ToString();
        txtDecimalLength.Text = frmDemo.Default.GrdView1.Column(frmDemo.Default.GrdView1.CurrentColIndex).DecimalLength.ToString();
    }


    private void cmdOK_Click(object sender, EventArgs e)
    {
        frmDemo.Default.GrdView1.AutoRedraw = false;
        if (txtColumnTitle.Enabled)
        {
            frmDemo.Default.GrdView1.Cell(0, frmDemo.Default.GrdView1.CurrentColIndex).Text = txtColumnTitle.Text;
        }
        frmDemo.Default.GrdView1.Column(frmDemo.Default.GrdView1.CurrentColIndex).FormatString = txtColumnFormat.Text;
        frmDemo.Default.GrdView1.Column(frmDemo.Default.GrdView1.CurrentColIndex).MaxLength = int.Parse(txtMaxLength.Text);
        frmDemo.Default.GrdView1.Column(frmDemo.Default.GrdView1.CurrentColIndex).DecimalLength = short.Parse(txtDecimalLength.Text);
        frmDemo.Default.GrdView1.AutoRedraw = true;
        this.Close();
    }


    private void cmdCancel_Click(object sender, EventArgs e)
    {
        this.Close();
    }
}
